datatypes/src/primitives/time_interval.rs

```diff
@@ -14,8 +14,8 @@ use std::fmt::{Debug, Display};
 #[derive(Clone, Copy, Deserialize, Serialize, PartialEq, Eq, ToSql, FromSql)]
 #[repr(C)]
 pub struct TimeInterval {
-    start: TimeInstance,
-    end: TimeInstance,
+    pub start: TimeInstance,
+    pub end: TimeInstance,
```

Comment:
You can't do this. `TimeInterval` ensures the invariant `start <= end`, so you can't just make it modifiable.

Reply:
c57310d69674b43e41291f85aaf0e2b5cda060e3